### PR TITLE
`edn-rs/edn-rs` <- `Grinkers/edn-rs`

### PR DESCRIPTION
**Pull request by Grinkers**
Issue discussion: 
https://github.com/edn-rs/edn-rs/issues/139

### CODE REVIEW

**Comment by Grinkers** [comment]
https://github.com/Grinkers/edn-rs/pull/2

**Comment by Grinkers** [comment]
edn-rs now passes roundtrip with clojure (jvm)'s own edn fuzzer
https://github.com/Grinkers/edn-rs/pull/3

**Comment by Grinkers** [comment]
Did a --force rebase with https://github.com/edn-rs/edn-rs/pull/130 (just to keep things clean).

**Comment by Grinkers** [comment]
trait Serialize uses reference to self instead of move
https://github.com/Grinkers/edn-rs/pull/4

**Comment by Grinkers** [comment]
Edn::Rational from String to (i64, u64)
https://github.com/Grinkers/edn-rs/pull/5

**Comment by Grinkers** [comment]
EdnError rework with parse rewrite
https://github.com/Grinkers/edn-rs/pull/6

File: src/edn/mod.rs [hunk 1]
```diff
@@ -536,7 +540,7 @@ impl Edn {
     /// use edn_rs::edn::{Edn, Vector};
     ///
     /// let edn = Edn::Vector(Vector::new(vec![Edn::Int(5), Edn::Int(6), Edn::Int(7)]));
-    /// let expected = "[5, 6, 7, ]";
+    /// let expected = "[5 6 7]";
```
Comment (naomijub) [inline src/edn/mod.rs:543]: Both are valid right?

File: src/edn/mod.rs [hunk 2]
```diff
@@ -780,15 +719,6 @@ mod test {
     use alloc::vec;
 
     use super::*;
-    #[test]
```
Comment (naomijub) [inline src/edn/mod.rs]: do we have these tests covered elsewhere?

Comment (Grinkers) [reply src/edn/mod.rs]: I removed them because with `(i64, u64)` they're infallable

**Comment by Grinkers** [comment]
@naomijub Thanks for starting to look at this. Instead of dealing with this huge draft PR, with some details inside that we may not want to upstream without discussion, I'll start breaking them apart piece by piece and squashing as we go. I mostly put this here for me to track.

We'll have to do it linearly though, I don't want to rebase/isolate 15 separate times; it's work built on top of other work. For example, Rational can be looked at in isolation, but after some other things.

**Comment by Grinkers** [comment]
Thinking about it, I'll start rebasing against edn-rs/main and deleting comments as we go. Think of this as more of an automatic linear issue tracker. Will delete when there's nothing left!

**Pull request closed without merging**